cmd/clusterctl/client/cluster/topology_plan_test.go

```diff
@@ -0,0 +1,492 @@
+/*
+Copyright 2022 The Kubernetes Authors.
```

Comment:
I would suggest either:
* renaming this file to topology_test.go
* or renaming topology.go to topology_plan.go

(So that the x_test.go file contains the unit tests for x.go)